pandas/io/formats/format.py

```diff
@@ -1121,39 +1117,22 @@ def _format_strings(self):
         return fmt_values.tolist()
 
 
-class IntervalArrayFormatter(GenericArrayFormatter):
-
-    def __init__(self, values, *args, **kwargs):
-        GenericArrayFormatter.__init__(self, values, *args, **kwargs)
-
-    def _format_strings(self):
-        formatter = self.formatter or str
-        fmt_values = np.array([formatter(x) for x in self.values])
-        return fmt_values
-
-
-class PeriodArrayFormatter(IntArrayFormatter):
-
+class ExtensionArrayFormatter(GenericArrayFormatter):
     def _format_strings(self):
-        from pandas.core.indexes.period import IncompatibleFrequency
-        try:
-            values = PeriodIndex(self.values).to_native_types()
-        except IncompatibleFrequency:
-            # periods may contains different freq
-            values = Index(self.values, dtype='object').to_native_types()
-
-        formatter = self.formatter or (lambda x: '{x}'.format(x=x))
-        fmt_values = [formatter(x) for x in values]
-        return fmt_values
-
+        values = self.values
+        if isinstance(values, (ABCIndexClass, ABCSeries)):
+            values = values._values
 
-class CategoricalArrayFormatter(GenericArrayFormatter):
+        formatter = values._formatter(boxed=True)
 
-    def __init__(self, values, *args, **kwargs):
-        GenericArrayFormatter.__init__(self, values, *args, **kwargs)
+        if is_categorical_dtype(values.dtype):
+            # Categorical is special for now, so that we can preserve tzinfo
+            array = values.get_values()
+        else:
+            array = np.asarray(values)
 
-    def _format_strings(self):
-        fmt_values = format_array(self.values.get_values(), self.formatter,
+        fmt_values = format_array(array,
```

Comment:
@TomAugspurger : i'm struggling to resolve some formatting issues. what is the reason for calling `format_array` here. As far as I can tell is looping back round to create a `GenericArrayFormatter` instance with a `formatter` specified to pick up the display options.

Reply:
i guess, to be more succinct, why is `super()._format_strings()` not used?

Reply:
I am not that familiar with this code, but from a quick look: calling `super()._format_strings()` would be different, as this would call `GenericArrayFormatter._format_strings`, while the generic `format_array` can still result in using custom formatters like `Datetime64(TZ)Formatter` or `Timedelta64Formatter`, depending on what the values of the underlying EA are.

Reply:
Although that most of those custom Formatter classes don't do much special *if* formatter is specified. 

Eg `Datetime64Formatter` has this in `_format_strings`:

https://github.com/pandas-dev/pandas/blob/181f972d2f4e6bd5c8c33721175f68188cfa1a0d/pandas/io/formats/format.py#L1174-L1175

Reply:
so if `ExtensionArrayFormatter` is not inheriting from `GenericArrayFormatter` but calling `format_array` to dispatch to another `...ArrayFormatter` class, why wouldn't the logic in `ExtensionArrayFormatter` be in `format_array`?